grammar.js

```diff
@@ -318,6 +318,11 @@ module.exports = grammar({
     [
       $.parenthesized_proc_type,
     ],
+
+    // Due to private / protected calls
+    [
+      $.call,
```

Comment:
I'd like to avoid adding `call` to `conflicts`. We've avoided that so far with precedence and careful rule construction. Having `call` conflict with itself could hide future precedence problems. 

I don't think this represents a legitimate conflict (where multiple rules _may_ apply, and the parser has to read farther to distinguish). I'm going to look for alternate solutions.

Reply:
That's valid. I meant to come back to this and take a look to see if I could figure it out, so I probably should've made it a draft PR. Help would be appreciated as you're more familiar with tree sitter than me.

Reply:
I ended up with 2 approaches:
https://github.com/crystal-lang-tools/tree-sitter-crystal/pull/41
https://github.com/crystal-lang-tools/tree-sitter-crystal/pull/42

#41 seems like the "right" way to use the tree-sitter features, but I don't like the increase in complexity.

I like #42 better overall, but it's a little bit of a hack and I left a comment on that PR about a possible drawback.

Reply:
Looking at both versions, #42 seems to be an overall better solution to this – #41 while correct is doing more processing of the syntax tree in regards to precedence. That in itself isn't a big issue, but it means there are multiple points of failure.

Reply:
Agreed, let's go with #42